packages/table/src/headers/header.tsx

```diff
@@ -213,13 +215,22 @@ export interface IHeaderState {
     hasSelectionEnded?: boolean;
 }
 
+const RESET_CELL_KEYS_BLACKLIST = [
+    "endIndex",
+    "shallowlyComparablePropKeysList",
+    "startIndex",
+    "style",
+    "viewportRect",
+] as Array<keyof IInternalHeaderProps>;
```

Comment:
we should expand https://palantir.github.io/tslint/rules/no-object-literal-type-assertion/ to cover this kind of thing